### PR TITLE
[PORT] Crate shelf from shiptest

### PR DESCRIPTION
**Pull request by warface1234455**
* port https://github.com/shiptest-ss13/Shiptest/pull/2374

# Document the changes in your pull request
You can now make crate shelf and put crates on it

# Why is this good for the game?
I believe that adding more options for organization is a good thing.
Specifically, this will ideally make players more likely to sort items into crates, rather than just dumping everything in.
# Testing
crate

# Spriting
image

# Wiki Documentation
Crate shelf, can be made using 1 metal, can store 3 crates maximum

# Changelog

:cl:  warface1234455, Generic DM, PositiveEntropy
rscadd: You can now make crate shelf and put crates on it
/:cl:

### CODE REVIEW

File: code/game/objects/structures/crateshelf.dm [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+	var/capacity = DEFAULT_SHELF_CAPACITY
+	var/use_delay = DEFAULT_SHELF_USE_DELAY
+	var/offset = DEFAULT_SHELF_VERTICAL_OFFSET
+	var/list/shelf_contents
+
+/obj/structure/crate_shelf/tall
+	capacity = 12
+
+/obj/structure/crate_shelf/Initialize()
```
Comment (iamthesilly) [inline code/game/objects/structures/crateshelf.dm]:
```suggestion
/obj/structure/crate_shelf/Initialize(mapload)
```

**Review by SapphicOverload — CHANGES_REQUESTED** [review]

**Comment by neeshacark** [comment]
generic left us and made shelf :(

**Comment by ShoziOffical** [comment]
> generic left us and made shelf :(

Yog sad

**Review by cuackles — APPROVED** [review]

File: code/game/objects/structures/tables_racks.dm [hunk 2]
```diff
@@ -652,14 +653,17 @@
...
 
 /obj/item/rack_parts/attack_self(mob/user)
+	if(locate(construction_type) in get_turf(user))
+		balloon_alert(user, "no room!")
+		return
 	if(building)
 		return
 	building = TRUE
-	to_chat(user, span_notice("You start constructing a rack..."))
+	to_chat(user, "<span class='notice'>You start assembling [src]...</span>")
```
Comment (iamthesilly) [inline code/game/objects/structures/tables_racks.dm]:
```suggestion
	to_chat(user, span_notice("You start assembling [src]..."))
```

**Review by SapphicOverload — CHANGES_REQUESTED** [review]

**Review by SapphicOverload — APPROVED** [review]

**Pull request merged**